utils/results/conversion/summaryparser/summaryparser.go

```diff
@@ -221,77 +249,124 @@ func getCveIds(cves []formats.CveRow, issueId string) []string {
+		}
+		sc.currentScan.Violations.Watches = utils.UniqueUnion(sc.currentScan.Violations.Watches, getJasScansWatches(secrets...)...)
+	}
+	return results.ApplyHandlerToJasIssues(results.ScanResultsToRuns(secrets), sc.entitledForJas, sc.getJasHandler(jasutils.Secrets, isViolationsResults))
 }
 
-func (sc *CmdResultsSummaryConverter) ParseIacs(_ results.ScanTarget, iacs ...*sarif.Run) (err error) {
+func (sc *CmdResultsSummaryConverter) ParseIacs(_ results.ScanTarget, isViolationsResults bool, iacs []results.ScanResult[[]*sarif.Run]) (err error) {
 	if !sc.entitledForJas || sc.currentScan.Vulnerabilities == nil {
 		// JAS results are only supported as vulnerabilities for now
 		return
 	}
 	if err = sc.validateBeforeParse(); err != nil {
 		return
 	}
-	if sc.currentScan.Vulnerabilities.IacResults == nil {
+	if !isViolationsResults && sc.currentScan.Vulnerabilities.IacResults == nil {
 		sc.currentScan.Vulnerabilities.IacResults = &formats.ResultSummary{}
 	}
-	return results.PrepareJasIssues(iacs, sc.entitledForJas, sc.getJasHandler(jasutils.IaC))
+	if isViolationsResults {
+		if sc.currentScan.Violations.IacResults == nil {
+			sc.currentScan.Violations.IacResults = &formats.ResultSummary{}
+		}
+		sc.currentScan.Violations.Watches = utils.UniqueUnion(sc.currentScan.Violations.Watches, getJasScansWatches(iacs...)...)
+	}
+	return results.ApplyHandlerToJasIssues(results.ScanResultsToRuns(iacs), sc.entitledForJas, sc.getJasHandler(jasutils.IaC, isViolationsResults))
 }
 
-func (sc *CmdResultsSummaryConverter) ParseSast(_ results.ScanTarget, sast ...*sarif.Run) (err error) {
+func (sc *CmdResultsSummaryConverter) ParseSast(_ results.ScanTarget, isViolationsResults bool, sast []results.ScanResult[[]*sarif.Run]) (err error) {
 	if !sc.entitledForJas || sc.currentScan.Vulnerabilities == nil {
 		// JAS results are only supported as vulnerabilities for now
 		return
 	}
 	if err = sc.validateBeforeParse(); err != nil {
 		return
 	}
-	if sc.currentScan.Vulnerabilities.SastResults == nil {
+	if !isViolationsResults && sc.currentScan.Vulnerabilities.SastResults == nil {
 		sc.currentScan.Vulnerabilities.SastResults = &formats.ResultSummary{}
 	}
-	return results.PrepareJasIssues(sast, sc.entitledForJas, sc.getJasHandler(jasutils.Sast))
+	if isViolationsResults {
+		if sc.currentScan.Violations.SastResults == nil {
+			sc.currentScan.Violations.SastResults = &formats.ResultSummary{}
+		}
+		sc.currentScan.Violations.Watches = utils.UniqueUnion(sc.currentScan.Violations.Watches, getJasScansWatches(sast...)...)
+	}
+	return results.ApplyHandlerToJasIssues(results.ScanResultsToRuns(sast), sc.entitledForJas, sc.getJasHandler(jasutils.Sast, isViolationsResults))
 }
 
-func (sc *CmdResultsSummaryConverter) getJasHandler(scanType jasutils.JasScanType) results.ParseJasFunc {
+func (sc *CmdResultsSummaryConverter) getJasHandler(scanType jasutils.JasScanType, violations bool) results.ParseJasFunc {
```

Comment:
I'd add a description about what exactly this handler is doing. it seems that it counts and returns a map with counts of SOMETHING but it is unclear what

Reply:
done, look again at the function, it does not return any map. it must first set the `count` map to the specific scan map based on the input `scanType` and `violations` after it caches the right map from the `currentScan` object it just adds one to the current value in it. I changed the comments so it will be more descriptive